backend/src/main/java/storage/repository/GenericRepository.java

```diff
@@ -0,0 +1,240 @@
+            }
+            query.append(key);
+        }
+        query.append(")");
+
+        for(int i = 0; i < jsonObject.keySet().size(); i++){
+            query.append(i==0?" VALUES (":i==1?"?":",?"); //skip ID
+        }
+        query.append(")");
+
+        try {
+            PreparedStatement ps = getPreparedStatementFromJSON(jsonObject, query); //maps model data to the query
+            ps.executeUpdate();
+            ResultSet rs = ps.getGeneratedKeys();
+            rs.next();
+            int id = rs.getInt(1);
+            return this.getObject(id);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Updates a model already in the system, currently updates ALL fields no matter how many are set (probably)
+     * @param object
+     * @return
+     */
+    public T updateObject(T object) {
+        JsonObject jsonObject = this.gson.toJsonTree(object).getAsJsonObject();
+        StringBuilder query = new StringBuilder("UPDATE "+ type.getSimpleName()+" SET");
+
+        for (String key : jsonObject.keySet()){
+            if(Objects.equals(key, "id"))continue; //skip id
+            String last = query.substring(query.length() - 1);
+            query.append(
+                last.equals("T")?
+                    " "+key :
+                    " = ?, "+key);
+        }
+        query.append(" = ? ");
+        query.append( " WHERE id="+object.getId());
+
+        try {
+            PreparedStatement ps = getPreparedStatementFromJSON(jsonObject, query); //maps model data to the query
+            ps.executeUpdate();
+            ResultSet rs = ps.getGeneratedKeys();
+            rs.next();
+            int id = rs.getInt(1);
+            return this.getObject(id);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Deletes a specific model instance
+     * @param id
+     * @return
+     */
+    public T deleteObject(int id) {
+        String query = "DELETE * FROM "+ type.getSimpleName() +" WHERE id="+id;
+        System.out.println("Running query: "+query);
+        ResultSet rs =  this.dbInterface.executeQuerry(query);
+        try {
+            return this.getMatchJSON(rs).iterator().next();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Attempts to convert result set to a model instance
+     * @param rs the result set from a querry
+     * @return An Object instance of given generic type
+     * @throws SQLException if error with result set
+     */
+    private List<T> getMatchJSON(ResultSet rs) throws SQLException {
+
+        List<T> results = new ArrayList<T>();
+        T result = null;
+        while(rs.next()){
+            try {
+                result = type.newInstance();
+            } catch (InstantiationException e) {
+                e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)